libevmasm/Assembly.cpp

```diff
@@ -679,13 +679,22 @@ LinkerObject const& Assembly::assemble() const
 		// Append an INVALID here to help tests find miscompilation.
 		ret.bytecode.push_back(static_cast<uint8_t>(Instruction::INVALID));
 
-	for (auto const& [subIdPath, bytecodeOffset]: subRef)
 	{
-		bytesRef r(ret.bytecode.data() + bytecodeOffset, bytesPerDataRef);
-		toBigEndian(ret.bytecode.size(), r);
-		ret.append(subAssemblyById(subIdPath)->assemble());
+		std::map<Assembly const*, size_t> subAssemblyOffsets;
```

Comment:
This should really be based on the keccak hash of the bytecode, not the assembly pointer.

Reply:
Is that a `std::` qualifier I see? UNACCEPTABLE